test/test_intensify/test_successive_halving.py

```diff
@@ -971,3 +972,148 @@ def test_incumbent_selection_designs(self):
         inc = intensifier._compare_configs(incumbent=self.config4, challenger=self.config3,
                                            run_history=self.rh, log_traj=False)
         self.assertEqual(self.config3, inc)
+
+    def test_count_running_instances_for_challenger(self):
+        """
+        Makes sure we return the proper number of running configs
+        """
+        def target(x):
+            return 1
+        taf = ExecuteTAFuncDict(ta=target, stats=self.stats, run_obj='quality')
+
+        taf.runhistory = self.rh
+        # select best on any budget
+        intensifier = SuccessiveHalving(
+            stats=self.stats, traj_logger=None,
+            rng=np.random.RandomState(12345), run_obj_time=False,
+            instances=list(range(20)), initial_budget=3, max_budget=10, eta=2)
+
+        # Get and process the first instance pair of config 1
+        intent, run_info = intensifier.get_next_run(
+            challengers=[self.config1, self.config2, self.config3, self.config4],
+            chooser=None,
+            run_history=self.rh,
+            incumbent=None,
+        )
+        result = eval_challenger(run_info, taf, self.stats, self.rh)
+        inc, inc_value = intensifier.process_results(
+            run_info=run_info,
+            incumbent=self.config1,
+            run_history=self.rh,
+            time_bound=np.inf,
+            result=result,
+            log_traj=False,
+        )
+        self.assertEqual(intensifier._count_running_instances_for_challenger(self.rh), 0)
+
+        # This will get us the second instance of config 1
+        intent, run_info = intensifier.get_next_run(
+            challengers=[self.config2, self.config3, self.config4],
+            chooser=None,
+            run_history=self.rh,
+            incumbent=None,
+        )
+
+        # Mark this run_info object as running and make sure that the intensifier can
+        # access it
+        self.rh.add(config=run_info.config,
+                    instance_id=run_info.instance,
+                    seed=run_info.seed,
+                    budget=run_info.budget,
+                    cost=10,
+                    time=1,
+                    status=StatusType.RUNNING,
+                    additional_info=None)
+        self.assertEqual(intensifier._count_running_instances_for_challenger(self.rh), 1)
+
+        # Now, when the intensifier is requested, to provide a new run,
+        # it should provide a new instance, not the same one
+        new_intent, new_run_info = intensifier.get_next_run(
+            challengers=[self.config2, self.config3, self.config4],
+            chooser=None,
+            run_history=self.rh,
+            incumbent=None,
+        )
+        self.assertNotEqual(new_run_info, run_info)
```

Comment:
If it should be a new instance, could you please check for exactly that?

Reply:
Improved check to check for same config and a new instance